onnxruntime/core/session/custom_ops.cc

```diff
@@ -784,14 +788,32 @@ ONNX_NAMESPACE::OpSchema CreateSchema(const std::string& domain, const OrtCustom
       }
     }
 
-    const auto type = op->GetInputType(op, i);
-    if (type == ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINED) {
-      undefined++;
+    std::unordered_set<ONNXTensorElementDataType> all_types;
+    for (auto o : ops) {
+      ORT_ENFORCE(i < o->GetInputTypeCount(o), "Another version of operator '", schema.Name(), "'has less inputs.");
+      const auto type = o->GetInputType(o, i);
+      if (type == ONNX_TENSOR_ELEMENT_DATA_TYPE_UNDEFINED) {
+        all_types.clear();
```

Comment:
comment explaining this a little would help. not clear to me why hitting an 'undefined' on one particular operator results in clearing `all_types`.

Reply:
I added a comment to explain. It is a convention I chose. Not empty to decalre precise types, empty to declare all of them.